cli/cmd/sync_v3.go

```diff
@@ -207,7 +233,29 @@ func syncConnectionV3(ctx context.Context, source v3source, destinations []v3des
 	isStateBackendEnabled := sourceSpec.BackendOptions != nil && sourceSpec.BackendOptions.TableName != ""
 
 	// Read from the sync stream and write to all destinations.
-	totalResources := 0
+	totalResources := int64(0)
+
+	var remoteProgressReporter *godebouncer.Debouncer
+	if progressAPIClient != nil {
+		teamName, syncName, syncRunId := os.Getenv("_CQ_TEAM_NAME"), os.Getenv("_CQ_SYNC_NAME"), os.Getenv("_CQ_SYNC_RUN_ID")
+		if teamName == "" || syncName == "" || syncRunId == "" {
+			return fmt.Errorf("failed to get team, sync or sync run from environment variables. got team: %s, sync name: %s, sync run ID: %s", teamName, syncName, syncRunId)
+		}
+		syncRunUUID, err := uuid.Parse(syncRunId)
+		if err != nil {
+			return fmt.Errorf("failed to parse sync run ID: %w", err)
+		}
+		remoteProgressReporter = godebouncer.NewWithOptions(godebouncer.WithTimeDuration(10*time.Second), godebouncer.WithTriggered(func() {
+			res, err := progressAPIClient.CreateSyncRunProgressWithResponse(ctx, teamName, syncName, syncRunUUID, cloudquery_api.CreateSyncRunProgressJSONRequestBody{Rows: totalResources})
+			if err != nil {
+				log.Warn().Err(err).Msg("Failed to send sync progress to API")
```

Comment:
```suggestion
				log.Warn().Err(err).Msg("Failed to send sync progress to API")
				return
```
Otherwise `res` will probably be `nil` and `res.StatusCode()` below will panic.

Reply:
Nice catch